### ISSUE TITLE
While Looping - Recursive Truncation

### ISSUE THREAD

**Issue by KDCinfo** [BeSite/jQuery.dotdotdot#72]
Wondering if anyone else has seen an issue with entire text blocks disappearing. This appears to be intermittent across browsers and mobile devices, but likely just in how they handle or count white space such as tab characters. It looks to be an issue with text chunks with less than 60 characters, but I cannot confirm this as of yet.

In stepping through the code (using Chrome debugger), I'm seeing it hit the following code loop, truncating the same text multiple times (although the text is truncated and looks to be correct with the first pass):

    while ( startPos <= endPos && !( startPos == 0 && endPos == 0 ) )

Initial Text: **Samsung Washer Dryer Bundle Deal just $0,000.00!**

After 1st pass: **Samsung Washer...**

After 2nd pass: **Samsung...**

After 3rd pass: **Sa...**

After 4th pass: **...**

This dangling ellipse then gets removed after another pass or two with:

    $e.detach();

I'm seeing this happen on various text blocks, but again, I'm really only seeing it with the smaller text chunks (less than 60 chars).

Although I'm still looking into this, thank you in advance for any thoughts or insights.

-KeithDC

**Comment by KDCinfo** [comment]
I just found my test conditions need to be narrowed down for FireFox and other browsers/devices. The below doesn't work directly on FireFox (it removes the entire text on even the longer text strings).

The following is using Chrome emulator iPhone 6. In troubleshooting this, I'm using some generic <span> tags so I can focus on just working and non-working text blocks:

    <span class="item-title-test">59 12345 67890 12345 67890 12345 67890 12345 67890xx xx xxx</span>
    <span class="item-title-test">60 12345 67890 12345 67890 12345 67890 12345 67890xx xx xxxx</span>

Executed in the Chrome debugger console:
    $('.item-title-test').dotdotdot();

**Comment by jameswilson** [comment]
Would be nice to see this setup in a jsbin or codepen.

**Comment by KDCinfo** [comment]
That's a really good suggestion, @jameswilson ... Perhaps that will help to expose if there is a setup issue on our side. I've started setting it up (a codepen stub), but will need to continue looking into it deeper, as I don't have my work code with me. Will post with any updates when I can. Thanks again.

**Comment by KDCinfo** [comment]
Well, I tried creating both a CodePen and jsFiddle stub, but neither are executing the dotdotdot. I'm new to both of these, so unsure if I may be doing something wrong or missing something.

CodePen: http://codepen.io/KeithDC/pen/aOBKzB/
jsFiddle: http://jsfiddle.net/KeithDC/pxa9gmkd/1/
